### DISCUSSION TITLE
[Feature] Show & Hide Faces per photo

### DISCUSSION THREAD

**Discussion by JesseWebDotCom** [immich-app/immich#5135] [Feature Request]
### The feature

Show & Hide Faces is great on the people tab as it helps you quickly clean up several faces in a few seconds. Can you add the same functionality when viewing individual photos? As a parent, I take photos at lots of events with unwanted faces (ex. kids sporting events, parties, outings, etc). Having this feature per photo will make it a lot easier to hide all the unwanted faces.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by martabal** [comment]
This is on my todo list but this is not really "show & hide". The idea is to un-assign the face from a person.

**Reply by JesseWebDotCom** [reply]
That's fantastic because I'll be in a situation where I have somewhat like 10 faces on a photo (maybe all are unwanted, maybe some) and it would be nice to quickly ignore the unwanted. 

Basically I'm saying the functionality on the people tab is also needed on each photo.  Here's another example: It would also be nice to assign individual faces on first view of a photo (where multiple faces are shown) - instead I right click on the face I want, open in new tab, and then assign a name (unlike the people tab where I can click the 3 dots and assign a name).

**Comment by jdf-ifdtech** [comment]
When photo stacking is implemented, it would be a great feature to have the option/button once clicked onto a photo to hide all people that aren't already matched in a photo. This would be most useful for a photo of someone infront of a crowd with multiple photos taken. Once the main subjects are found, you don't need to hide the remaining background people individually.

**Comment by gcarver** [comment]
Got done importing my pictures a few days ago and a particular set of crowd shots at my brothers graduation led to a few hundred "people". They were pretty randomly shown on the /people page so hiding them was very haphazard. I thought a better approach would be to select a problematic photo and remove people from there. I probably didn't save any time overall but I wrote a tampermonkey script to add a "Hide this person" button to the root people page. In case anyone finds it useful, the script is at https://gist.github.com/gcarver/76ced7efb0c6e7429b54e3c7b859e32c

To use, open a photo from the timeline with a bunch of people you want to hide. Faces will appear on the right side of the screen (info tray). Bulk open unwanted faces in a bunch of new tabs (ctrl+click for me). Switch to one of the tabs. Click the big "Hide this person" button in the toolbar. The window automatically closes. Another person tab will likely get focus next. Repeat until they're all hidden and you're back to the original photo. Refresh and confirm you've hidden everyone you wanted. The workflow is not amazing, but it's quicker -- saving a bunch of mouse work and page navigation.

**Comment by chodthewacko** [comment]
The only big thing I need at this point is a way to hide all faces for any given pictures, to make group/crowd pictures manageable.
I'd prefer 'hide all unnamed faces', but I'd settle for the option to hide all faces, and then manually unhide the 1-2 faces I care about.
Far better than the dozens and dozens of clicks currently needed to manually hide each face.   (and no, auto-hiding faces with only X pictures doesn't work here)
